File: 06/assembler/src/tecs/assembler/AsmPattern.java
package tecs.assembler;

class AsmPattern {
		// @100 or @varname
	static final String AT = "\\s*@(\\S+).*";

	// 100
	static final String NUMBER = "^\\d+.*";

	// D=A or D=A;JMP or A;JMP (COMP part not optional)
	static final String COMP = "\\s*([DMA]{1,3})?\\s*=?\\s*([DMA\\-+&\\|!01]{1,3})\\s*;?\\s*([A-Z]{3})?.*";

	// (LABELNAME)
	static final String LABEL = "\\s*\\((\\S+)\\).*";
}

File: 06/src/tecs/assembler/AsmPattern.kt
package tecs.assembler

object AsmPattern {
    // @100 or @varname
    val AT = """\s*@(\S+).*""".toRegex()

    // 100
    val NUMBER = """^\d+.*""".toRegex()

    // D=A or D=A;JMP or A;JMP (COMP part not optional)
    val COMPUTE = """\s*([DMA]{1,3})?\s*=?\s*([DMA\-+&|!01]{1,3})\s*;?\s*([A-Z]{3})?.*""".toRegex()

    // (LABELNAME)
    val LABEL = """\s*\((\S+)\).*""".toRegex()
}